form-view-model/src/test/kotlin/com/ritense/formviewmodel/validation/OnStartUpViewModelValidatorTest.kt

```diff
@@ -0,0 +1,150 @@
+package com.ritense.formviewmodel.validation
+
+import com.ritense.form.domain.FormIoFormDefinition
+import com.ritense.form.service.impl.FormIoFormDefinitionService
+import com.ritense.formviewmodel.BaseTest
+import com.ritense.formviewmodel.submission.FormViewModelStartFormSubmissionHandlerFactory
+import com.ritense.formviewmodel.submission.FormViewModelUserTaskSubmissionHandlerFactory
+import com.ritense.formviewmodel.submission.TestStartFormSubmissionHandler
+import com.ritense.formviewmodel.viewmodel.TestViewModel
+import com.ritense.formviewmodel.viewmodel.TestViewModelLoader
+import com.ritense.formviewmodel.viewmodel.ViewModel
+import com.ritense.formviewmodel.viewmodel.ViewModelLoader
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.mockito.InjectMocks
+import org.mockito.Mock
+import org.mockito.Mockito.mock
+import org.mockito.kotlin.doReturn
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.whenever
+import java.io.ByteArrayOutputStream
+import java.io.PrintStream
+import java.util.Optional
+import kotlin.reflect.KClass
+import kotlin.test.assertTrue
+
+class OnStartUpViewModelValidatorTest : BaseTest() {
+
+    @Mock
+    private lateinit var formIoFormDefinitionService: FormIoFormDefinitionService
+
+    @InjectMocks
+    private lateinit var onStartUpViewModelValidator: OnStartUpViewModelValidator
+    private lateinit var viewModelLoaders: List<ViewModelLoader<*>>
+    private lateinit var viewModelLoader: ViewModelLoader<ViewModel>
+    private lateinit var formViewModelStartFormSubmissionHandlerFactory: FormViewModelStartFormSubmissionHandlerFactory
+    private lateinit var formViewModelUserTaskSubmissionHandlerFactory: FormViewModelUserTaskSubmissionHandlerFactory
+
+    @BeforeEach
+    fun setUp() {
+        formIoFormDefinitionService = mock()
```

Comment:
This is some limbo between using `@Mock` vs `mock()`. I prefer using constructor injection so that we also get rid of the `lateinit var` stuff.

Reply:
This is an opinion and the nature of open source projects should not be to have 1 view imo. These are minor things imo. To argue for mock() is that @InjectMocks is kinda magic. So I rather use mock() over @Mock. 
Contribution guidelines should then be more clear? what do you think?.

Reply:
This specific comment was about the `parseViewModel` being annotated with `@Mock`, and then set again in the `setup()`. Same with the `@InjectMocks`. Since `lateinit var` can only be set once, the `@Mock` and `@InjectMocks` annotation clearly aren't working properly. Probably because the class is not annotated with `@ExtendWith(MockitoExtension.class)`. Hence the 'limbo'. 

You are right that the rest of it (constructor injection) is personal preference.

Reply:
Will remove the annotation